src/server/pps/cmds/cmds.go

```diff
@@ -1797,34 +1817,74 @@ func pipelineHelper(ctx context.Context, pachctlCfg *pachctl.Config, reprocess b
 			log.Error(ctx, "problem adding trace data", zap.Error(err))
 		}
 
+		var spec map[string]any
+		if err := json.Unmarshal([]byte(specJSON), &spec); err != nil {
+			return errors.Errorf("could not decode %s as JSON", specJSON)
+		}
+
 		if pushImages {
-			if request.Transform == nil {
+			if spec["transform"] == nil {
 				return errors.New("must specify a pipeline `transform`")
 			}
-			if err := dockerPushHelper(request, registry, username); err != nil {
+			if err := dockerPushHelper(spec, registry, username); err != nil {
 				return err
 			}
 		}
 
-		if request.Pipeline.Project.GetName() == "" {
-			request.Pipeline.Project = &pfs.Project{Name: project}
+		pipeline, ok := spec["pipeline"].(map[string]any)
+		if !ok {
+			return errors.New("must specify pipeline, an object")
+		}
+		project, ok := pipeline["project"]
+		if !ok {
```

Comment:
all of these checks seem like they should be handled by decoding/encoding

Reply:
The values can be nil, or (in the YAML) of different types.